### ISSUE TITLE
[Bug]: Requests And Limits set in the Custom Serving Runtimes template get ignored

### ISSUE THREAD

**Issue by bdattoma** [opendatahub-io/odh-dashboard#1775]
### ODH Component

Model Serving

### Current Behavior

Custom Serving Runtime can have Limits And Requests set, but these are ignored when creating a model server. User is requested to select the model server size out of the default sizes (small, medium, etc) or inserting custom values.

Moreover, the default Serving Runtimes have Requests and Limits set to 0 which can me confusing

### Expected Behavior

If the Custom Serving Runtime has requests and limits set, these should be reported in the Model Server creation modal when the runtime is chosen.

### Steps To Reproduce

1. Create a Custom Serving Runtime from Dashboard > Settings > Serving Runtime
2. Create a DS Project
3. Start creation of a model server
4. Select the Serving Runtime created at step 1)

**Comment by lucferbux** [comment]
I'm gonna update my thoughts here cause i couldn't attend the UX meeting.
I would say that the resources selected in the Model Server modal will always override the ones in the Custom Serving Runtime spec, at the end of the day the Custom Serving Runtimes are only blueprints.
I would agree that we might wanna have some sort of logic to adapt to our UI, so if a user select a Custom Serving Runtime in the Model Server modal, we automatically select the resources/limits in the spec but **then** the user can freely use those or override them.
What do you think @vconzola ?

**Comment by vconzola** [comment]
@lucferbux I guess my question is, "Is that how we want it to work?" Is this an architecture question that @jwforres should weigh in on? Also, if the serving runtimes are truly templates, should the user even be able to edit them, or should they be view only? And how does all this fit in with what Chris Chase has been talking about forever with the "curly braces" - I still don't understand what that means? Users should not have parts of the serving runtime that they can change and the changes stick, while other parts that they change can be overridden by UI settings. That would be too confusing.

**Comment by andrewballantyne** [comment]
@vconzola Well let us be clear here... Custom Serving Runtimes are just the template that starts the ServingRuntime (aka the server) in the Project. This has not ever been 1:1 and cannot be 1:1 today as there is no customization of the ServingRuntime at that level.

The `curly brace` conversation Chris (and I) was having is for us to embrace the OpenShift Templates we are using to supply a configurable nature to the create server modal. Essentially for us to allow the admin to "ask specific questions" of the user in the Project view when they host a server. Instead of it always being "give us server size", "give us accelerator".

Today, until we support proper parameterization of the Custom Serving Runtimes (which we avoided in our first pass), the templates are nothing more than "starting points" for the Project view -- which the best we can do is "prefill" the modal with some information.

If we support the Templates fully and give them param usage (which will also give them the ability to deploy multiple resources along with the Serving Runtime) the admin can say "no, you cannot specify the resource limits", or maybe just "requests". The admin will be able to say "you can supply me with x, y, and z, the rest is static". So once we enable parameters the admin can lock out the user from specifying anything they don't want. We're just going to read and react. This is somewhat an over simplification, as I imagine there are some special use-cases to this, but effectively that's how it will work.

So the question isn't "should we do that" it's a question of "when are we supporting it" and if we will wait for the full adoption of the Templates before we do that.

The real UX question here is "Do we do this at all now or do we wait for templates to be properly supported with params".

**Comment by vconzola** [comment]
> This has not ever been 1:1 and cannot be 1:1 today as there is no customization of the ServingRuntime at that level.
@andrewballantyne I don't understand what you mean here.

> The real UX question here is "Do we do this at all now or do we wait for templates to be properly supported with params".
I don't think this is a UX question - I think it's a BU question. I think we all agree params is our ultimate destination. But what's the priority and how long can the product wait for them? In the meantime, if we're going to allow end user input to override admin settings for resource requests and limits (also replicas and GPUs?) should we at least inform the admins this is possible?

**Comment by andrewballantyne** [comment]
I think there are assumptions being made here about what the Custom Serving Runtimes were supposed to cover.

Perhaps we should talk less about what it "is" and talk more about "what we thought it was" or "what we want it to be". Because development _always_ thought of these as "starting points" not "overriding things the user can do" -- leaving that to be "params will control what can and what cannot change". We have a YAML editor, with almost no guidance on what it means to build a Custom ServingRuntime. On face value, it's "set this thing up" and then "go find out what it does in the Project layer".

> This has not ever been 1:1 and cannot be 1:1 today as there is no customization of the ServingRuntime at that level.
> @andrewballantyne I don't understand what you mean here.

I mean that we never prescribed the flow. It was _always_ (as I saw it) a "starting point" and at Project usage time we put in the values for the `container` and any other items we needed. There was never a thought of "well if admin sets up the containers we should respect them" -- because we always believed the wide lockdown and choices would be covered by params.

> From a UX standpoint, this seems like a bug

This is where my messages are coming from -- this is not a bug as the flow is very open ended and we have had no guidance to setup this flow in any other way than we have done so (the "start something on the admin side" and "finish it on the project side" design).

If we want to change the flow -- we can. But parameterization also solves this more specifically and gives the admin the power to choose what is configurable and what isn't -- resources/container values as well as any part of their ServingRuntime spec.

If I had to make a call today, I'd close this issue as won't do and wait for proper parameterization. Since no matter how we solve this, the question of "what do we support" will always be put into question. Parameters solve this by saying "you configure what you configure" by the specification of the params not by some hidden set of requirements.

Maybe we need to discuss this more in a meeting before we do anything.
